### PR TITLE
add support for Fiery Brand mitigation

### PR DESCRIPTION
**Pull request by ToppleTheNun**
image
image

### CODE REVIEW

File: src/parser/shared/modules/Enemies.ts [hunk 1]
```diff
@@ -72,6 +73,42 @@ class Enemies extends Entities<Enemy> {
...
 
+  /**
+   * Retrieves the enemy that was the source of given event.
+   * @param {object} event
+   * @returns {Enemy|null}
+   */
+  getSourceEntity(event: AnyEvent): Enemy | null {
+    if (!HasTarget(event)) {
+      return null;
+    }
```
Comment (kfinch) [inline src/parser/shared/modules/Enemies.ts]: `!HasSource(event)) {` instead?

Comment (ToppleTheNun) [reply src/parser/shared/modules/Enemies.ts]: Good catch!

File: src/parser/shared/modules/Enemies.ts [hunk 2]
```diff
@@ -72,6 +73,42 @@ class Enemies extends Entities<Enemy> {
...
+  getSourceEntity(event: AnyEvent): Enemy | null {
+    if (!HasSource(event)) {
+      return null;
+    }
+    if (event.sourceIsFriendly) {
+      return null;
+    }
+    if (!event.sourceID) {
+      return null;
+    }
```
Comment (kfinch) [inline src/parser/shared/modules/Enemies.ts]: `!HasSource(event)` and `!event.sourceID` are redundant

Comment (ToppleTheNun) [reply src/parser/shared/modules/Enemies.ts]: Also makes the `"sourceInstance" in event` check unnecessary

**Review by kfinch — APPROVED** [review]

**Pull request merged**